### PR TITLE
CallKit opt out

### PR DESCRIPTION
**Pull request by michaelkirk**
based on #1683 

Allows opting out of call kit on iOS10.

### CODE REVIEW

File: Signal/src/call/UserInterface/CallUIAdapter.swift [hunk 1]
```diff
@@ -76,11 +76,11 @@ extension CallUIAdaptee {
             // So we use the non-CallKit call UI.
             Logger.info("\(TAG) choosing non-callkit adaptee for simulator.")
             adaptee = NonCallKitCallUIAdaptee(callService: callService, notificationsAdapter: notificationsAdapter)
-        } else if #available(iOS 10.0, *) {
+        } else if #available(iOS 10.0, *), Environment.getCurrent().preferences.isCallKitEnabled() {
```
Comment (charlesmchen) [inline Signal/src/call/UserInterface/CallUIAdapter.swift]: Am I misreading the swift syntax?  Why is there a comma and not a &&

File: Signal/src/view controllers/AdvancedSettingsTableViewController.m [hunk 2]
```diff
@@ -101,26 +120,61 @@ - (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger
...
+    AdvancedSettingsTableViewControllerSection settingsSection = (AdvancedSettingsTableViewControllerSection)section;
+    switch (settingsSection) {
+        case AdvancedSettingsTableViewControllerSectionLogging:
+            return NSLocalizedString(@"LOGGING_SECTION", nil);
+        case AdvancedSettingsTableViewControllerSectionCalling:
+            return NSLocalizedString(@"SETTINGS_SECTION_TITLE_CALLING", @"settings topic header for table section");
+        case AdvancedSettingsTableViewControllerSectionPushNotifications:
+            return NSLocalizedString(@"PUSH_REGISTER_TITLE", @"Used in table section header and alert view title contexts");
+        default:
+            return 0;
```
Comment (michaelkirk) [inline Signal/src/view controllers/AdvancedSettingsTableViewController.m]: return nil, and log?

File: Signal/src/call/UserInterface/CallUIAdapter.swift (hunk 1 above)
Comment (michaelkirk) [inline Signal/src/call/UserInterface/CallUIAdapter.swift]: welcome to swift. =/

**Comment by michaelkirk** [comment]
in https://github.com/WhisperSystems/Signal-iOS/commit/2a9aa4c852584120c762a3bf32166b77eabc2b27

**Pull request closed without merging**